packages/datadog-instrumentations/src/process.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const shimmer = require('../../datadog-shimmer')
+const { channel } = require('dc-polyfill')
+
+const startSetUncaughtExceptionCaptureCallback = channel('datadog:process:setUncaughtExceptionCaptureCallback:start')
+
+if (process.setUncaughtExceptionCaptureCallback) {
+  let currentCallback
+
+  shimmer.wrap(process, 'setUncaughtExceptionCaptureCallback',
+    function wrapSetUncaughtExceptionCaptureCallback (originalSetUncaughtExceptionCaptureCallback) {
+      return function setUncaughtExceptionCaptureCallback (newCallback) {
+        if (startSetUncaughtExceptionCaptureCallback.hasSubscribers) {
+          const abortController = new AbortController()
+          startSetUncaughtExceptionCaptureCallback.publish({ newCallback, currentCallback, abortController })
+          if (abortController.signal.aborted) {
+            return
+          }
+        }
+
+        const result = originalSetUncaughtExceptionCaptureCallback.apply(this, arguments)
+
+        currentCallback = newCallback
+
+        return result
```

Comment:
apply could fail? if not, why not change the order?

```suggestion
        currentCallback = newCallback

        return originalSetUncaughtExceptionCaptureCallback.apply(this, arguments)
```

Reply:
yes, could fail. 
This will faill, for example
```javascript
process.setUncaughtExceptionCaptureCallback(fn1)
process.setUncaughtExceptionCaptureCallback(fn2)
```

But this will work
```javascript
process.setUncaughtExceptionCaptureCallback(fn1)
process.setUncaughtExceptionCaptureCallback(null)
process.setUncaughtExceptionCaptureCallback(fn2)
```